cpp/test/core/blob_tests.cpp

```diff
@@ -33,6 +33,80 @@ TEST_CASE_BEGIN(LargeBlob)
 }
 TEST_CASE_END
 
+TEST_CASE_BEGIN(BlobMoveAssign)
+{
+    using bond::blob;
+
+    // workaround: old compilers won't let us print nullptr, while new ones won't
+    // compare nullptrs to ints
+    void* null = 0;
+
+    const uint32_t size = 32;
+    struct Big
+    {
+        char big[size];
+    };
+
+    // move default-constructed blob
+    {
+        auto data = boost::make_shared<Big>();
+        blob a{data, size};
+        blob b;
+
+        UT_AssertAreEqual(data.use_count(), 2);
+
+        a = std::move(b);
+        
+        UT_AssertAreEqual(b.data(), data.get());
+        UT_AssertAreEqual(a.size(), 0u);
+        UT_AssertAreEqual(a.data(), null);
+        UT_AssertAreEqual(a.content(), null);
+        UT_AssertAreEqual(data.use_count(), 2);
+    }
+    // move from instance with shared ptr
+    {
+        // move from other instance
+        auto data = boost::make_shared<Big>();
+        blob a;
+        blob b{data, size};
+
+        UT_AssertAreEqual(data.use_count(), 2);
+
+        a = std::move(b);
+
+        UT_AssertAreEqual(a.data(), data.get());
+        UT_AssertAreEqual(a.size(), size);
+        UT_AssertAreEqual(b.data(), null);
+        UT_AssertAreEqual(b.content(), null);
+        UT_AssertAreEqual(data.use_count(), 2);
+    }
+    // move from raw memory
+    {
+        std::unique_ptr<Big> p1(new Big);
+        std::unique_ptr<Big> p2(new Big);
+        blob a{p1.get(), size};
+        blob b{p2.get(), size};
+        blob c;
+
+        a = std::move(b);
+        UT_AssertAreEqual(a.data(), p2.get());
+        UT_AssertAreEqual(a.size(), size);
+
+        // assert more than is guaranteed - we check if move is implemented with swap
+        UT_AssertAreEqual(b.data(), p1.get());
+        UT_AssertAreEqual(b.size(), size);
+
+        // move to empty blob
+        c = std::move(a);
+        UT_AssertAreEqual(c.data(), p2.get());
+        UT_AssertAreEqual(c.size(), size);
+        // our implementation does clear the source, but the standard only requires it must be 
+        // in a valid state
+        UT_AssertAreEqual(a.data(), null);
```

Comment:
Can you have a helper function `check_blob_is_empty` which checks for `data`, `size` and `empty`. Standard requires us to keep the object in a state where it can be safely destroyed (in which case the implementation could be just `=default`), but here we want to make the moved-from blob to be empty.